apidump/apidump.go

```diff
@@ -631,12 +655,22 @@ func Run(args Args) error {
 			sig := make(chan os.Signal, 2)
 			signal.Notify(sig, os.Interrupt)
 			signal.Notify(sig, syscall.SIGTERM)
-			select {
-			case received := <-sig:
-				printer.Stderr.Infof("Received %v, stopping trace collection...\n", received.String())
-			case err := <-errChan:
-				stopErr = err
-				printer.Stderr.Errorf("Encountered error while collecting traces, stopping...\n")
+
+			// Continue until an interrupt or all collectors have stopped with errors.
+		DoneWaitingForSignal:
+			for {
+				select {
+				case received := <-sig:
+					printer.Stderr.Infof("Received %v, stopping trace collection...\n", received.String())
+					break DoneWaitingForSignal
+				case interfaceErr := <-errChan:
+					printer.Stderr.Errorf("Encountered an error on interface %s, continuing with remaining interfaces.  Error: %s\n", interfaceErr.interfaceName, interfaceErr.err.Error())
```

Comment:
This will print even when there are no remaining interfaces, which might be confusing. I don't think this is a blocker for the PR, though.

Reply:
Good idea, fixed in [7fd0f98](https://github.com/akitasoftware/akita-cli/pull/140/commits/7fd0f988c09858ee5c7225564ca7153d9e154cfe).